src/Main.java

```diff
@@ -0,0 +1,363 @@
+        printTwoDimensionalArray(givenArray);
+        if (indexOfLine == -1 && indexOfElementInLine == -1) {
+            throw new NoElementInArray("Данное число отсутствует в массиве");
+        }
+        else {
+            if (indexOfLine < givenArray.length - 1 && indexOfElementInLine < givenArray[indexOfLine].length - 1 && indexOfLine != 0 && indexOfElementInLine != 0) {
+                multiply = givenArray[indexOfLine-1][indexOfElementInLine-1] *
+                        givenArray[indexOfLine-1][indexOfElementInLine+1] *
+                        givenArray[indexOfLine+1][indexOfElementInLine-1] *
+                        givenArray[indexOfLine+1][indexOfElementInLine+1];
+            }
+            else if (indexOfLine == 0 && indexOfElementInLine == 0) {
+                multiply = givenArray[indexOfLine+1][indexOfElementInLine+1];
+            }
+            else if (indexOfLine == givenArray.length-1 && indexOfElementInLine == 0) {
+                multiply = givenArray[indexOfLine-1][indexOfElementInLine+1];
+            }
+            else if (indexOfLine == givenArray.length-1 && indexOfElementInLine == givenArray[indexOfLine].length-1) {
+                multiply = givenArray[indexOfLine-1][indexOfElementInLine-1];
+            }
+            else if (indexOfLine == 0 && indexOfElementInLine == givenArray[indexOfLine].length-1) {
+                multiply = givenArray[indexOfLine+1][indexOfElementInLine-1];
+            }
+            else if (indexOfLine == 0 && indexOfElementInLine < givenArray[indexOfLine].length-1) {
+                multiply = givenArray[indexOfLine+1][indexOfElementInLine-1] * givenArray[indexOfLine+1][indexOfElementInLine+1];
+            }
+            else if (indexOfLine < givenArray.length-1 && indexOfElementInLine == givenArray[indexOfLine].length-1) {
+                multiply = givenArray[indexOfLine+1][indexOfElementInLine-1] * givenArray[indexOfLine-1][indexOfElementInLine-1];
+            }
+            else if (indexOfLine == givenArray.length-1 && indexOfElementInLine < givenArray[indexOfLine].length-1) {
+                multiply = givenArray[indexOfLine-1][indexOfElementInLine-1] * givenArray[indexOfLine-1][indexOfElementInLine+1];
+            }
+            else if (indexOfLine < givenArray.length-1 && indexOfElementInLine == 0) {
+                multiply = givenArray[indexOfLine-1][indexOfElementInLine+1] * givenArray[indexOfLine+1][indexOfElementInLine+1];
+            }
+            return multiply;
+        }
+
+
+    }
+
+    public static boolean hasDatePassed(Holidays holiday) {
+        Calendar calendar = new GregorianCalendar();
+        long currentDateMillis = calendar.getTimeInMillis();
+        long holidayDateMillis = holiday.getCalendar().getTimeInMillis();
+        return currentDateMillis > holidayDateMillis;
+    }
+
+    public static boolean hasDatePassed2(Holidays2 holiday) {
+        LocalDate currentDate = LocalDate.now();
```

Comment:
there is LocalDate  method for it: isAfter, isBefore
you don't need to implement it yourself